.github/workflows/build-nginx-proxy-image.yml

```diff
@@ -0,0 +1,174 @@
+          docker-compose up -d test-multi-proxy
+
+          until docker inspect tests_test-multi-proxy_1 --format='{{.State.Health.Status}}' | grep -q healthy; do
+            sleep 2
+          done
+
+          echo "Test for location 1 @ /..."
+          curl --fail -sk -o /dev/null https://localhost:10443/
+          echo "Test for payment page..."
+          curl --fail -sk -o /dev/null -H "Host: www.payments.service.gov.uk" https://localhost:10443/using-govuk-pay
+
+          docker-compose stop test-multi-proxy
+
+      - name: Test Multi-Location, Single Proxy and NAXSI Config
+        shell: bash
+        run: |
+          cd ${GITHUB_WORKSPACE}/tests
+          docker-compose up -d test-multi-location-naxsi
+
+          until docker inspect tests_test-multi-location-naxsi_1 --format='{{.State.Health.Status}}' | grep -q healthy; do
+            sleep 2
+          done
+
+          echo "Test for all OK..."
+          curl --fail -sk -o /dev/null -H "Host: www.payments.service.gov.uk" https://localhost:10443/
+
+          docker-compose stop test-multi-location-naxsi
+
+      - name: Test Custom Upload Size Config
+        shell: bash
+        run: |
+          cd ${GITHUB_WORKSPACE}/tests
+          docker-compose up -d test-custom-upload
+
+          until docker inspect tests_test-custom-upload_1 --format='{{.State.Health.Status}}' | grep -q healthy; do
+            sleep 2
+          done
+
+          echo "Generate a random test file..."
+          dd if=/dev/urandom of=/tmp/bigfile.txt bs=1048576 count=10
+
+          echo "Upload the large test file..."
+          curl -sk -o /tmp/upload_test.txt -F "file=@/tmp/bigfile.txt;filename=nameinpost" https://localhost:10443/uploads/doc
+
+          echo "Grepping the response..."
+          grep "Thanks for the big doc" /tmp/upload_test.txt &> /dev/null
+
+          docker-compose stop test-custom-upload
+
+      - name: Test Static File Cache Setting
+        shell: bash
+        run: |
+          cd ${GITHUB_WORKSPACE}/tests
+          docker-compose up -d test-static-cache
+
+          until docker inspect tests_test-static-cache_1 --format='{{.State.Health.Status}}' | grep -q healthy; do
+            sleep 2
+          done
+
+          echo "Test for all OK..."
+          curl -s -I -X GET -k --compressed https://localhost:10443/gzip | grep -q 'Content-Encoding: gzip'
+
+          docker-compose stop test-static-cache
+
+      - name: Test Big Buffers Setting
+        shell: bash
+        run: |
+          cd ${GITHUB_WORKSPACE}/tests
+          docker-compose up -d test-big-buffers
+
+          until docker inspect tests_test-big-buffers_1 --format='{{.State.Health.Status}}' | grep -q healthy; do
+            sleep 2
+          done
+
+          echo "Test for all OK..."
+          curl -s -I -X GET -k --compressed https://localhost:10443/gzip | grep -q 'Content-Encoding: gzip'
+
+          docker-compose stop test-big-buffers
+
+      - name: Shutdown Docker Compose Stack
```

Comment:
I wouldn't bother with the shutdown in GHA, once the workflow ends everythign is thrown away anyway, doing the shutdown just means extra time waiting for the workflow to complete